tests/ops/op_math/test_controlled_ops.py

```diff
@@ -428,3 +435,105 @@ def test_unitary_check(self):
             qml.ControlledQubitUnitary(
                 not_unitary, control_wires=[0, 2], wires=1, unitary_check=True
             )
+
+
+NON_PARAMETRIZED_OPERATIONS = [
+    (qml.CY, CY),
+]
+
+
+class TestOperations:
+    @pytest.mark.parametrize("op_cls, _", NON_PARAMETRIZED_OPERATIONS)
+    def test_nonparametrized_op_copy(self, op_cls, _, tol):
+        """Tests that copied nonparametrized ops function as expected"""
+        op = op_cls(wires=0 if op_cls.num_wires is AnyWires else range(op_cls.num_wires))
+        copied_op = copy.copy(op)
+        np.testing.assert_allclose(op.matrix(), copied_op.matrix(), atol=tol)
+
+    @pytest.mark.parametrize("ops, mat", NON_PARAMETRIZED_OPERATIONS)
+    def test_matrices(self, ops, mat, tol):
+        """Test matrices of non-parametrized operations are correct"""
+        op = ops(wires=0 if ops.num_wires is AnyWires else range(ops.num_wires))
+        res_static = op.compute_matrix()
+        res_dynamic = op.matrix()
+        assert np.allclose(res_static, mat, atol=tol, rtol=0)
+        assert np.allclose(res_dynamic, mat, atol=tol, rtol=0)
+
+
+class TestDecompositions:  # pylint: disable=too-few-public-methods
+    def test_CY_decomposition(self, tol):
+        """Tests that the decomposition of the CY gate is correct"""
+        op = qml.CY(wires=[0, 1])
+        res = op.decomposition()
+
+        mats = []
+        for i in reversed(res):
+            if len(i.wires) == 1:
+                mats.append(np.kron(i.matrix(), np.eye(2)))
+            else:
+                mats.append(i.matrix())
+
+        decomposed_matrix = np.linalg.multi_dot(mats)
```

Comment:
```suggestion
        decomposed_matrix = qml.matrix(op.decomposition)()
```

A hacky little trick you can do so you don't have to kron matrices together manually.

In addition to testing the matrix, it'd be best to test the actual contents of the decomposition as well using `qml.equal(op1, op2)`.

Reply:
Good to know. Thanks :-)

Reply:
I included [tests](https://github.com/PennyLaneAI/pennylane/blob/move-CY-to-controlled-ops/tests/ops/op_math/test_controlled_ops.py#L472) for the individual components of the decomposition now. Does it look good to you?